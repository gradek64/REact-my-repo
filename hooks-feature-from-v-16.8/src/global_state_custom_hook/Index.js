import React from 'react';

import Counter1 from './Components/CounterComponent_1';
import Counter2 from './Components/ConterCompoent_2';

const CombineTogether = () => (
  <div>
    <Counter1 />
    <Counter1 />
  </div>
);

export default CombineTogether;
