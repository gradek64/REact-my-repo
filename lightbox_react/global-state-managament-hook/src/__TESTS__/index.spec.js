import React from 'react';
import { render } from 'react-dom';
import { shallow } from 'enzyme';
import Index from '../index';

//mock dom render
jest.mock('react-dom', () => ({
  render: jest.fn(() => {}),
}));
//custom
import customEvent from '../customEvents/dispatchEvent';

//spy on customEvent.emit
jest.spyOn(customEvent, 'emit').mockImplementation(() => {});

describe('Index', () => {
  it('should render correctly', () => {
    const component = shallow(<Index />);
    expect(component).toMatchSnapshot();
  });

  it('when button 1 is clicked it shoudl call customEvent.emit method', () => {
    const wrapper = shallow(<Index />);
    const button1 = wrapper.find('button').first();
    button1.simulate('click');
    expect(customEvent.emit).toBeCalledWith('OPEN_MODAL_1');
  });

  it('when button 2 is clicked it shoudl call customEvent.emit method', () => {
    const wrapper = shallow(<Index />);
    const button2 = wrapper.find('button').last();
    button2.simulate('click');
    expect(customEvent.emit).toBeCalledWith('OPEN_MODAL_2');
  });
});
