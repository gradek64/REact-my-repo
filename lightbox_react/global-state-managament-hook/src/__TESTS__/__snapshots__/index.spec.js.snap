// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render correctly 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    modal 1
  </button>
  <button
    onClick={[Function]}
  >
    modal 2
  </button>
  <Modal
    eventOn="OPEN_MODAL_1"
  >
    <p>
      this is modal test
    </p>
  </Modal>
  <Modal
    eventOn="OPEN_MODAL_2"
  >
    <TestComponent />
  </Modal>
</Fragment>
`;
